<a href="https://colab.research.google.com/github/bbevan/Colabs/blob/master/Opp_Deleted_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introduction
Microsoft Excel is a capable data analysis tool with a mature codebase and a well-known brand. However, over the course of the last few years, Data Analysts, Executives, and Computer Programmers have shyed away from Excel. As the overall tech market has leaned towards Open Source technology, Analysts have followed by utilizing Open Source software to perform data analysis.

This is an example of a Jupyter notebook (a front end for performing data analysis). It gives Analysts access to the Python programming language, which is equipped with a number of code libraries geared towards data ingestion, cleaning, manipulation, transformation, modeling, and visualization. Using Python in the Jupyter notebook, I have performed a number of analyses on the American Red Cross's National Recruitment data.


# The Data
The Data was gathered from the "My Reports" section of Volunteer Connection. For various reasons, it's often difficult for the server to generate a report in Excel format, possibly due to server load or slow code. I was able to gather the Data from June 2018 to January 2019 in CSV format (comma-separated values), and then I performed a cleaning procedure in order to properly format the data. 



In [0]:
# Load up Requirements
import pandas as pd
import seaborn as sns
import numpy as np

* Above: an example of a code cell that contains instructions for performing the analyses.

I have stored the CSV file inside my Google Drive for ease of access. It has been turned into an Excel spreadsheet however, and it is accessible via[ this sharepoint link.](https://americanredcross-my.sharepoint.com/:x:/g/personal/brandon_bevan_redcross_org/Ebj5FJGiUgZIiKirGnDYAtUBs1nnU0QNLGbJiY2-2zuR3g?e=afLC3G)

In [2]:
# Mount Google Drive
from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/gdrive


* Drive was mounted at 12:08 PM on 1/28/2019

With the Pandas library, the CSV can be read and accessed by the Python programming language.

In [6]:
# Load CSV into `data` variable
data = pd.read_csv('/content/gdrive/My Drive/June to January clean.csv')

# Print the data dimensions
data.shape

(9694, 30)

* Data loaded successfully
* 9694 rows × 30 columns

In [8]:
# Look at the first 5 rows
data.head()

Unnamed: 0        Service Name*  \
0           0  Biomedical Services   
1           1  Biomedical Services   
2           2  Biomedical Services   
3           3  Biomedical Services   
4           4  Biomedical Services   

                           Global Position Name Recruiting  \
0  NHQ:BIO - Diverse Donor Outreach Team Member         No   
1  NHQ:BIO - Diverse Donor Outreach Team Member         No   
2  NHQ:BIO - Diverse Donor Outreach Team Member         No   
3  NHQ:BIO - Diverse Donor Outreach Team Member         No   
4  NHQ:BIO - Diverse Donor Outreach Team Member         No   

                      Division Name                       Region Name    Type  \
0  ARC National Operations Division           ARC National Operations  Global   
1               Crossroads Division  Greater Cincinnati-Dayton Region  Global   
2               Crossroads Division                    Indiana Region  Global   
3               Crossroads Division                    Indiana Region  Global   
4               Crossroads Division                   Kentucky Region  Global   

                            Local Position Name Auto-Refer To Name  \
0  NHQ:BIO - Diverse Donor Outreach Team Member   Edwards, Vincent   
1  NHQ:BIO - Diverse Donor Outreach Team Member   Edwards, Vincent   
2  NHQ:BIO - Diverse Donor Outreach Team Member   Edwards, Vincent   
3  NHQ:BIO - Diverse Donor Outreach Team Member   Edwards, Vincent   
4  NHQ:BIO - Diverse Donor Outreach Team Member   Edwards, Vincent   

      Account Name        ...        # Days Sub-Ref Days Pending Approved Dt  \
0     Adam, Dawuda        ...                   NaN          NaN         NaN   
1  Holbrook, Allie        ...                   NaN          NaN         NaN   
2  Kilbride, Eilis        ...                   NaN          NaN         NaN   
3    Espich, Megan        ...                  51.0          NaN         NaN   
4     Moss, Rachel        ...                   NaN          NaN         NaN   

  # Days Ref-Appr Not Approv Dt # Days Ref-NotAppr  \
0             NaN       9/14/18                NaN   
1             NaN       8/25/18                NaN   
2             NaN       8/27/18                NaN   
3             NaN      11/26/18               66.0   
4             NaN      12/18/18                NaN   

  Completed Date (Appr/Not Appr) # Days Ref-Compl       Referred To  \
0                        9/14/18              NaN               NaN   
1                        8/25/18              NaN               NaN   
2                        8/27/18              NaN               NaN   
3                       11/26/18             66.0  Edwards, Vincent   
4                       12/18/18              NaN               NaN   

        Assigned To  
0               NaN  
1               NaN  
2               NaN  
3  Edwards, Vincent  
4               NaN  

[5 rows x 30 columns]

* The `head` function will display the first 5 rows in the Data.

# Analysis via Pivots
 
Most of the data is qualitative in nature, meaning non-numerical. However, using Pivots, we can create numerical summaries of the data by counting or summing certain fields within subcategories of the raw data. 

For example, we can count the number of `Opp Deleted` "Yes" tags by Auto-Referral names, Divisions, Regions, etc. We can also take tallies of the number of applications by Region and Division.

## Number of Applicants by Division

In [22]:
# Number of Referrals by Division
first_pivot = lambda a: pd.pivot_table(data, index=a, values=["Account Name"],aggfunc=len ).sort_values(by="Account Name", ascending=False)
first_pivot("Division Name")

Account Name
Division Name                                       
Pacific Division                                2042
Mid-Atlantic Division                           1943
Southeast and Caribbean Division                1728
Northeast Division                              1180
Southwest and Rocky Mountains Division           901
North Central Division                           842
Crossroads Division                              765
ARC National Operations Division                 172
Division 29 SAF Stations                         121

## Number of Applicants by Region

In [23]:
# Number of Referrals by Region
first_pivot("Region Name")

Account Name
Region Name                                       
National Capital Area Region                   515
Northern California Coastal Region             482
Georgia Region                                 428
New Jersey Region                              413
Los Angeles Region                             384
Eastern North Carolina Region                  376
Gold Country Region                            345
Central and South Texas Region                 344
Chicago and Northern Illinois Region           336
Central Florida Region                         294
Michigan Region                                268
South Florida Region                           258
Greater New York Region                        257
Northwest Region                               257
Greater Carolinas Region                       244
Greater Chesapeake Region                      203
Eastern Pennsylvania Region                    197
Tennessee Region                               192
Virginia Region                                185
South Carolina Region                          184
ARC National Operations                        172
North Florida Region                           169
Connecticut & Rhode Island Region              169
Massachusetts Region                           166
San Diego Region                               157
Indiana Region                                 146
Desert to the Sea Region                       142
Texas Gulf Coast Region                        133
Louisiana Region                               132
Central California Region                      128
...                                            ...
Dakotas Region                                  57
Minnesota Region                                56
Western and Central New York Region             53
West Virginia Region                            49
Ohio Buckeye Region                             41
Greater PA Region                               39
New Hampshire and Vermont Region                37
Alaska Region                                   27
Idaho and Montana Region                        27
Ramstein Region                                 26
Pacific Islands Region                          26
Maine Region                                    23
Central and Southern Illinois Region            19
Italy / Spain Region                            14
Camp Humphreys                                  13
Camp Arifjan                                    12
Djibouti                                         8
Camp Foster MCB                                  8
Yokosuka                                         7
Kadena Air Base                                  7
Stuttgart Region                                 5
US Naval Station Guantanamo Bay                  5
Kunsan Air Base                                  4
Atsugi                                           3
Camp Casey-Red Cloud                             3
Misawa Air Base                                  3
Yokota Air Base                                  3
Camp Zama                                        2
Iwakuni MCAS                                     2
Yongsan                                          1

[77 rows x 1 columns]

## Number of Applicants by Auto-Referral Name

In [24]:
# Number of Referrals by Auto-Referral name
first_pivot("Auto-Refer To Name")

Account Name
Auto-Refer To Name                 
Peoples, Michael               2007
Larson, Jamie                   971
Connor, Tammy                   962
Johnson, La Vinda               811
Kong, Bee                       546
Pinarbasi, Peggy                384
Sudol, Theodore                 375
Hooper, Peter                   329
Innocenzi, Mark                 313
Evans, Sam                      312
Borgmann, Kathryn               248
Domenico, Michelle              247
Bartman, Nicole                 230
Kurtinitis, Jennifer            146
De Luna, Angelita               124
Laakso, Betsy                   115
Edwards, Vincent                109
Sterrett, Mary Clare            106
Walker, Clara                    86
Snow-Nicosia, Rebecca            77
Aston, Esme                      76
Ellis, Leonid                    75
Ruiz, Sara                       73
Richards, Kasie                  73
Waananen, Susan                  72
Cassel, Stephen                  71
Rogers, Paula                    69
McDonnell, Joshua                65
Ryan, Eileen                     60
Pirovano, Peggy                  51
...                             ...
Dixon, Pamela                    24
Hatch, Larissa                   21
Dulgar, Megan                    18
Thompson, Patricia               17
Ferry, Griffin                   17
Watson, Susan                    13
Drenning, Anna                   13
Beckley, June                    13
Richards, Heather                12
Williams, Eric                   12
Kagel, Lisa                      10
Myers, Julie                     10
BOUDREAUX, LABARRON               7
Sykes, Stephanie                  6
Halsne, Jodie                     5
Vara, Wendy                       5
Myers, Laurynn                    4
Reiter, Josh                      3
MacIntyre, Linda                  3
Bottle, Brendon                   3
Villegas, Veronica                3
Baker, Jennifer                   2
Hodgers, Cristen                  2
Maltbie, Charles                  2
Whiteman, Jessica                 1
Putt, Kira                        1
D'Ambrosio, Maria                 1
Oriatti, Janet                    1
Gonzalez, David                   1
Zerivitz, Brad                    1

[62 rows x 1 columns]

## Number of Applicants by Global Position Name

In [26]:
# Number of Referrals by Global Position Name
first_pivot("Global Position Name")

Account Name
Global Position Name                                            
NHQ:VS - VIPC Intake Volunteer                               983
NHQ: 2018 Hurricane Season Customer Service Age...           490
NHQ:VS - Volunteer Screener                                  416
NHQ:HR - Talent Acquisition Volunteer (Biomedic...           404
NHQ:HR - Talent Acquisition Volunteer (Humanita...           293
NHQ:VS - Operations Support Volunteer                        278
NHQ:DCS - Disaster Training & Development Volun...           276
NHQ:CFO Units RES/FIN/IT - REFIT Region Steward              246
NHQ:ISD - Volunteer Partner to the Internationa...           233
NHQ:ISD - Restoring Family Links (RFL) Helpline...           201
NHQ:ISD - Volunteer Onboarding Lead                          182
NHQ:DCS - Recovery Services Long-Term Recovery ...           178
NHQ:IT - End User Services: IT Service Desk (IT...           165
NHQ:Finance - Disaster Services Finance - Finan...           159
NHQ:ISD - Latin American Outreach and Coordinat...           147
NHQ:HR - Talent Acquisition Volunteer (NHQ/Deve...           146
NHQ:DCS - Disaster Training Web-Based Course Te...           143
NHQ:DCS - Response Services NSS Remote Support ...           142
NHQ:Finance - Financial Shared Services - Accou...           141
NHQ:ISD - Volunteer Interviewer                              134
NHQ:ISD - Interpreter/Translator                             133
NHQ:VS - Volunteer Screener Lead                             130
NHQ:HR - Talent Acquisition Volunteer Engagemen...           125
NHQ:ISD - Restoring Family Links (RFL) Training...           121
NHQ:IT - End User Services: Volunteer Technical...           120
NHQ:DCS - AmeriCorps Administrative Support                  115
NHQ:ISD - Digital Tool Testing Volunteer                     114
NHQ:ISD - Graphic Design Volunteer: Global Prep...           110
NHQ:BIO - Diverse Donor Outreach Team Member                 109
NHQ:IT - Humanitarian IT - Volunteer Connection...           108
...                                                          ...
NHQ:VS - VIPC Hot Shot Team (HST) Member                       1
NHQ:ISD - Volunteer Partner to the Executive Di...             1
NHQ:SAF - Hero Care Casework Volunteer Supervisor              1
NHQ:SAF - Communications and Social Media Volun...             1
NHQ:Finance - Financial Shared Services - Suppl...             1
NHQ:Finance - Financial Shared Services - Suppl...             1
NHQ:Finance - Financial Shared Services - Suppl...             1
NHQ:Finance - Financial Shared Services - Volun...             1
NHQ:Finance - Humanitarian Finance - Program Su...             1
NHQ:HR - Learning & Development Volunteer                      1
NHQ:Finance - Disaster Services Finance - Finan...             1
NHQ:Finance - Business Planning & Decision Supp...             1
NHQ:ISD - Cash Transfer Programming Volunteer                  1
NHQ:ISD - Copy Editing/Document Improvement Vol...             1
NHQ:ISD - Diaspora Program Advisor Volunteer                   1
NHQ:FD - In-Kind Donations Volunteer                           1
NHQ:ISD - Graphic Design Volunteer                             1
NHQ:DCS - Response Services Program Support                    1
NHQ:DCS - Response Services NSS Field Remote Op...             1
NHQ:DCS - Response Services Mass Care Communica...             1
NHQ:ISD - Special Events Volunteer                             1
NHQ:ISD - Volunteer Connection Lead                            1
NHQ:Finance - Financial Operations - Finance An...             1
NHQ:DCS - Recovery Services Doctrine Support Vo...             1
NHQ:ISD - Volunteer Partner to the Regional Dir...             1
NHQ:DCS - Recovery Services CAN-Recovery Case I...             1
NHQ:DCS - RMS Mobilization Manager                             1
NHQ:ODIS - Diversity & Inclusion Assistant                     1
NHQ:OGC - Unpaid Internship                                    1
NHQ:VS - Youth Field A

# `Opp Deleted` Counts

One of the metrics we've spoken about is the `Opp Deleted` column. Let's try to break this down by pivot tables as well.

## Opp Deleted by Division

In [30]:
# Pivoting function
second_pivot = lambda a: pd.pivot_table(data, index=a, values=["Opp Deleted"],aggfunc=pd.DataFrame.count).sort_values(by="Opp Deleted", ascending=False)

second_pivot("Division Name")

Opp Deleted
Division Name                                      
Southeast and Caribbean Division                758
Pacific Division                                546
Southwest and Rocky Mountains Division          423
Mid-Atlantic Division                           222
Northeast Division                              182
Crossroads Division                             153
North Central Division                           69
ARC National Operations Division                 20
Division 29 SAF Stations                         20

## Opp Deleted by Region

In [31]:
second_pivot("Region Name")

Opp Deleted
Region Name                                      
Central Florida Region                        276
Central and South Texas Region                242
Northern California Coastal Region            222
Georgia Region                                197
South Carolina Region                         105
Northwest Region                               93
Desert to the Sea Region                       88
Greater Cincinnati-Dayton Region               64
Central California Region                      61
South Florida Region                           61
New Jersey Region                              60
North Florida Region                           57
Puerto Rico Region                             56
Louisiana Region                               53
Cascades Region                                52
Chicago and Northern Illinois Region           52
Alabama Region                                 52
Arizona New Mexico El Paso Region              51
Connecticut & Rhode Island Region              49
Colorado-Wyoming Region                        48
West Virginia Region                           45
Texas Gulf Coast Region                        44
National Capital Area Region                   44
Greater New York Region                        31
Oklahoma Region                                31
Eastern Pennsylvania Region                    29
New Hampshire and Vermont Region               29
Indiana Region                                 29
ARC National Operations                        20
Greater Carolinas Region                       17
...                                           ...
Greater PA Region                               3
Yokota Air Base                                 3
Minnesota Region                                3
Atsugi                                          3
Camp Humphreys                                  3
Misawa Air Base                                 3
Massachusetts Region                            2
Dakotas Region                                  2
Yokosuka                                        1
Ramstein Region                                 1
Tennessee Region                                1
Kentucky Region                                 1
Camp Casey-Red Cloud                            1
Kansas Nebraska - SW Iowa Region                1
Camp Zama                                       1
North Texas Region                              1
Northeast Ohio Region                           0
Djibouti                                        0
Camp Arifjan                                    0
Wisconsin Region                                0
Western and Central New York Region             0
Camp Foster MCB                                 0
Idaho and Montana Region                        0
Italy / Spain Region                            0
Iwakuni MCAS                                    0
Stuttgart Region                                0
Kunsan Air Base                                 0
Maine Region                                    0
Mississippi Region                              0
Yongsan                                         0

[77 rows x 1 columns]

## Opp Deleted by Auto-Referral Name

In [47]:
second_pivot("Auto-Refer To Name")

Opp Deleted
Auto-Refer To Name               
Peoples, Michael              498
Larson, Jamie                 286
Connor, Tammy                 192
Kong, Bee                     151
Johnson, La Vinda             134
Sudol, Theodore               131
Pinarbasi, Peggy              128
Hooper, Peter                  86
Borgmann, Kathryn              78
Evans, Sam                     76
Kurtinitis, Jennifer           62
Domenico, Michelle             48
Laakso, Betsy                  42
Innocenzi, Mark                37
Waananen, Susan                35
Ellis, Leonid                  30
Edwards, Vincent               29
Bartman, Nicole                27
Cassel, Stephen                26
Walker, Clara                  23
De Luna, Angelita              21
Rogers, Paula                  20
Richards, Kasie                19
Ruiz, Sara                     16
Pirovano, Peggy                14
Hansen, Grant                  13
Rose, Alex                     13
Dixon, Pamela                  12
Sterrett, Mary Clare           12
Aston, Esme                    11
...                           ...
McDonnell, Joshua               8
Hatch, Larissa                  7
Beckley, June                   7
Myers, Julie                    4
Kagel, Lisa                     3
Williams, Eric                  3
Thompson, Patricia              3
Dulgar, Megan                   3
Halsne, Jodie                   2
BOUDREAUX, LABARRON             1
MacIntyre, Linda                1
Watson, Susan                   1
Villegas, Veronica              1
Drenning, Anna                  1
Bottle, Brendon                 1
Richards, Heather               1
Hodgers, Cristen                0
Ryan, Eileen                    0
Oriatti, Janet                  0
D'Ambrosio, Maria               0
Putt, Kira                      0
Sykes, Stephanie                0
Myers, Laurynn                  0
Vara, Wendy                     0
Gonzalez, David                 0
Maltbie, Charles                0
Baker, Jennifer                 0
Whiteman, Jessica               0
Reiter, Josh                    0
Zerivitz, Brad                  0

[62 rows x 1 columns]

# Opp Deleted by Global Position Name

In [48]:
second_pivot("Global Position Name")

Opp Deleted
Global Position Name                                           
NHQ:VS - VIPC Intake Volunteer                              166
NHQ:HR - Talent Acquisition Volunteer (Biomedic...          105
NHQ:VS - Volunteer Screener                                 101
NHQ:HR - Talent Acquisition Volunteer (Humanita...           93
NHQ:DCS - Disaster Training & Development Volun...           90
NHQ: 2018 Hurricane Season Customer Service Age...           65
NHQ:VS - Operations Support Volunteer                        64
NHQ:DCS - Disaster Training Web-Based Course Te...           61
NHQ:ISD - Volunteer Partner to the Internationa...           57
NHQ:CFO Units RES/FIN/IT - REFIT Region Steward              55
NHQ:ISD - Volunteer Onboarding Lead                          55
NHQ:VS - Volunteer Screener Lead                             50
NHQ:ISD - Restoring Family Links (RFL) Helpline...           49
NHQ:DCS - Recovery Services Long-Term Recovery ...           48
NHQ:HR - Talent Acquisition Volunteer (NHQ/Deve...           48
NHQ:DCS - Response Services NSS Remote Support ...           46
NHQ:Finance - Disaster Services Finance - Finan...           45
NHQ:ISD - Volunteer Interviewer                              45
NHQ:DCS - AmeriCorps Administrative Support                  42
NHQ:DCS - Disaster Institute Program Administrator           41
NHQ:ISD - Latin American Outreach and Coordinat...           40
NHQ:ISD - Interpreter/Translator                             40
NHQ:HR - Talent Acquisition Volunteer Engagemen...           40
NHQ:Finance - Financial Shared Services - Accou...           39
NHQ:DCS - Response Services Feeding and DES Pro...           38
NHQ:DCS - Response Services Mass Care Project M...           32
NHQ:ISD - Graphic Design Volunteer: Global Prep...           30
NHQ:DCS - Recovery Fiscal Review Volunteer                   30
NHQ:BIO - Diverse Donor Outreach Team Member                 29
NHQ:VS - VIPC Regional Intake Lead                           26
...                                                         ...
NHQ:ISD - Diaspora Program Advisor Volunteer                  0
NHQ:ISD - Cash Transfer Programming Volunteer                 0
NHQ:ODIS - Diversity & Inclusion Assistant                    0
NHQ:Finance - Financial Shared Services - Gener...            0
NHQ:Finance - Financial Shared Services - Suppl...            0
NHQ:Finance - Financial Shared Services - Suppl...            0
NHQ:Finance - Financial Shared Services - Suppl...            0
NHQ:HR - Talent Acquisition Volunteer (Biomedic...            0
NHQ:Finance - Financial Shared Services - Volun...            0
NHQ:HR - Learning & Development Volunteer                     0
NHQ:Finance - Disaster Services Finance - Finan...            0
NHQ:Finance - Business Planning & Decision Supp...            0
NHQ:Finance - Biomedical Finance - Document Spe...            0
NHQ:ISD - Policy and Partnerships Volunteer                   0
NHQ:DCS - Response Services NSS Field Remote Op...            0
NHQ:IT - Humanitarian IT - Administrative Write...            0
NHQ:DCS - Response Services Program Support                   0
NHQ:IT - End User Services: Data Analyst Volunt...            0
NHQ:Finance - Humanitarian Finance - Program Su...            0
NHQ:DCS - Workforce Readiness Specialist                      0
NHQ:ISD - Volunteer Partner to the Regional Dir...            0
NHQ:FD - In-Kind Donations Volunteer                          0
NHQ:ISD - Volunteer Partner to the Executive Di...            0
NHQ:FD - Tiffany Circle National Council                      0
NHQ:ISD - Special Events Volunteer                            0
NHQ:ISD - Sharepoint Administrator                            0
NHQ:FD - Unpaid Internship                                    0
NHQ:ISD - Restoring Family Links (RFL) Administ...            0
NHQ:ISD - Recovery Services Volunteer                         0
NHQ:VS - Youth Field Ambassador                               0

[181 rows x 1 colum

# Total Counts

In [51]:
data.count()

Unnamed: 0                        9694
Service Name*                     9694
Global Position Name              9694
Recruiting                        9694
Division Name                     9694
Region Name                       9694
Type                              9694
Local Position Name               9694
Auto-Refer To Name                9506
Account Name                      9694
Current Status                    9694
Email Address                     9690
Status on Opp Submission          9636
Opp Deleted                       2393
Referral Status                   9694
Progress Status                   6871
Application Date                  7896
VIPC Passed to Region             6417
Submitted Dt                      9694
Referred Dt                       2456
# Days Sub-Ref                    2456
Days Pending                        94
Approved Dt                        814
# Days Ref-Appr                    762
Not Approv Dt                     6446
# Days Ref-NotAppr       

## Opp Deleted Counts

In [52]:
data.count()["Opp Deleted"]

2393

## Approved Counts

In [53]:
data.count()["Approved Dt"]

814

## Not Approved Counts

In [54]:
data.count()["Not Approv Dt"]

6446

# Summary

It does appear that 2393 applications were deleted between June 2018 and January 2019. 

I have created an Excel spreadsheet containing all of the deleted applications in the original data, which can be accessed via[ this sharepoint link](https://americanredcross-my.sharepoint.com/:x:/g/personal/brandon_bevan_redcross_org/EYf0P8l5Ix5LkcUZgu-LR3oB9Xfn6_WXTNJd3MJqbYXsAg?e=h8gJ6P).